pkg/apis/pipeline/v1/param_types.go

```diff
@@ -139,6 +139,16 @@ func (ps Params) extractParamMapArrVals() map[string][]string {
 	return paramsMap
 }
 
+// extractParamMapStrVals creates a param map with the key: param.Name and
+// val: param.Value.ArrayVal
+func (ps Params) extractParamMapStrVals() map[string]string {
+	paramsMap := make(map[string]string)
+	for _, p := range ps {
+		paramsMap[p.Name] = p.Value.StringVal
```

Comment:
Wondering if we can replace this func with just using the Params?

Reply:
The param mapping makes it easier to look up key/value pairs and merge mapping with Combinations using `maps.Copy()`